desc/geometry/curve.py

```diff
@@ -740,10 +771,82 @@ def compute(
             transforms=transforms,
             data=data,
             override_grid=override_grid,
-            basis_in=self._basis,
+            basis_in=self.basis,
             **kwargs,
         )
 
+    @classmethod
+    def from_values(cls, coords, N=10, basis="xyz", name=""):
+        """Fit coordinates to FourierPlanarCurve representation.
+
+        Parameters
+        ----------
+        coords: ndarray, shape (num_coords,3)
+            Coordinates to fit a FourierPlanarCurve object with each column
+            corresponding to xyz or rpz depending on the basis argument.
+        N : int
+            Fourier resolution of the new r representation.
+        basis : {"rpz", "xyz"}
+            Basis for input coordinates. Defaults to "xyz".
+        name : str
+            Name for this curve.
+
+        Returns
+        -------
+        curve : FourierPlanarCurve
+            New representation of the curve parameterized by a Fourier series for r.
+
+        """
+        # convert to xyz basis
+        if basis == "rpz":
+            coords = rpz2xyz(coords)
+        coords = np.atleast_2d(coords)
+
+        # center
+        center = np.mean(coords, axis=0)
+        coords = coords - center  # shift to origin
+
+        # normal
+        idx = coords.shape[0] // 2
+        normal = np.cross(coords[0, :], coords[idx, :])  # 2 arbitrary points on curve
```

Comment:
Could compute all the normals and take the average which might be better if the curve isn't perfectly planar